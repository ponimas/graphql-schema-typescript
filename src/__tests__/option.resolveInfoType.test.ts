import { executeApiTest } from './testUtils';

describe('resolveInfoType', () => {
    it('should default to GraphQLResolveInfo', async () => {
        const schemaStr = `schema { query: RootQuery } type RootQuery { test: String! }`;
        const generated = await executeApiTest('schemaString.ts', {}, schemaStr);
        expect(generated).toContain('GraphQLResolveInfo');
    });

    it('should use custom resolve info', async () => {
        const schemaStr = `schema { query: RootQuery } type RootQuery { test: String! }`;
        const generated = await executeApiTest('schemaString.ts', {
                                                   resolveInfoType: 'CustomResolveInfo',
                                                   importStatements: ['interface CustomResolveInfo {}'] // so ts compiles
                                               },
                                               schemaStr);
        expect(generated).toContain('CustomResolveInfo');
        expect(generated).not.toContain('GraphQLResolveInfo');
    });
});