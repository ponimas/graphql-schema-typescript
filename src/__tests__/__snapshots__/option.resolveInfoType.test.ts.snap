// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolveInfoType should default to GraphQLResolveInfo 1`] = `
"/* tslint:disable */
/* eslint-disable */
import { GraphQLResolveInfo } from 'graphql';
/**
 * This file is auto-generated by @ruiaraujo/graphql-schema-typescript
 * Please note that any changes in this file may be overwritten
 */
 

/*******************************
 *                             *
 *          TYPE DEFS          *
 *                             *
 *******************************/
export interface GQLRootQuery {
  test: string;
}

/*********************************
 *                               *
 *         TYPE RESOLVERS        *
 *                               *
 *********************************/
/**
 * This interface define the shape of your resolver
 * Note that this type is designed to be compatible with graphql-tools resolvers
 * However, you can still use other generated interfaces to make your resolver type-safed
 */
export interface GQLResolver {
  RootQuery?: GQLRootQueryTypeResolver;
}
export interface GQLRootQueryTypeResolver<TParent = any> {
  test?: RootQueryToTestResolver<TParent>;
}

export interface RootQueryToTestResolver<TParent = any, TResult = any> {
  (parent: TParent, args: {}, context: any, info: GraphQLResolveInfo): TResult;
}
"
`;

exports[`resolveInfoType should use custom resolve info 1`] = `
"/* tslint:disable */
/* eslint-disable */
interface CustomResolveInfo {}
/**
 * This file is auto-generated by @ruiaraujo/graphql-schema-typescript
 * Please note that any changes in this file may be overwritten
 */
 

/*******************************
 *                             *
 *          TYPE DEFS          *
 *                             *
 *******************************/
export interface GQLRootQuery {
  test: string;
}

/*********************************
 *                               *
 *         TYPE RESOLVERS        *
 *                               *
 *********************************/
/**
 * This interface define the shape of your resolver
 * Note that this type is designed to be compatible with graphql-tools resolvers
 * However, you can still use other generated interfaces to make your resolver type-safed
 */
export interface GQLResolver {
  RootQuery?: GQLRootQueryTypeResolver;
}
export interface GQLRootQueryTypeResolver<TParent = any> {
  test?: RootQueryToTestResolver<TParent>;
}

export interface RootQueryToTestResolver<TParent = any, TResult = any> {
  (parent: TParent, args: {}, context: any, info: CustomResolveInfo): TResult;
}
"
`;
